zk/arkworks/gramine/src/note/mod.rs

```diff
@@ -247,15 +249,140 @@ impl<'de> Deserialize<'de> for Note {
     }
 }
 
+
+trait ToCanonicalEncoding {
```

Comment:
Can we reuse the `CanonicalSerialize/CanonicalDeserealize` traits instead? (`Fq` already implements them)